src/string_pool.cr

```diff
@@ -1,19 +1,55 @@
+# A String pool is a collection of Strings.
+# It allows a runtime to save memory by preserving String in a pool.
+# We can reuse instance of common String instead of creating a new String.
+#
+#
+#   require "string_pool"
+#   pool = StringPool.new
+
 class StringPool
+  # Returns the size
+  #
+  #     pool = StringPool.new
+  #     pool.size # => 0
   getter size
 
+  # Initialization of the elements 
+  # At the begining, the size is null and the bucket is an array of size 11
```

Comment:
The size is not "null". There are no "elements". Maybe just say "Creates a new empty StringPool".